connection_notifier.cc

```diff
@@ -34,14 +34,23 @@ sstring to_string(client_type ct) {
     }
 }
 
+static sstring to_string(client_connection_stage ccs) {
+    switch (ccs) {
+        case client_connection_stage::established: return connection_stage_literal<client_connection_stage::established>;
+        case client_connection_stage::authenticating: return connection_stage_literal<client_connection_stage::authenticating>;
+        case client_connection_stage::ready: return connection_stage_literal<client_connection_stage::ready>;
+        default: throw std::runtime_error("Invalid client_connection_stage");
```

Comment:
Won't the compiler complain about switch not covering all the enum values?
I think adding default destroys this compile-time safety, no?
Probably better to throw after the switch.

Reply:
Done